From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Subject: kabi protect hnae_ae_ops
Patch-Mainline: never, kABI
References: bsc#1104353 FATE#326415

Commit 26483246b548 ("net: hns3: Add support for hns3_nic_netdev_ops.ndo_do_ioctl")
added field do_ioctl ti struct hnae3_ae_ops, which breaks kabi.
Fix this by moving the new field to the end of the struct and hide
it from kabi checker.

---
 drivers/net/ethernet/hisilicon/hns3/hnae3.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/hisilicon/hns3/hnae3.h
+++ b/drivers/net/ethernet/hisilicon/hns3/hnae3.h
@@ -344,8 +344,6 @@ struct hnae3_ae_ops {
 	void (*get_mac_addr)(struct hnae3_handle *handle, u8 *p);
 	int (*set_mac_addr)(struct hnae3_handle *handle, void *p,
 			    bool is_first);
-	int (*do_ioctl)(struct hnae3_handle *handle,
-			struct ifreq *ifr, int cmd);
 	int (*add_uc_addr)(struct hnae3_handle *handle,
 			   const unsigned char *addr);
 	int (*rm_uc_addr)(struct hnae3_handle *handle,
@@ -418,6 +416,10 @@ struct hnae3_ae_ops {
 			      unsigned long *supported,
 			      unsigned long *advertising);
 	void (*get_port_type)(struct hnae3_handle *handle, u8 *port_type);
+#ifndef __GENKSYMS__
+	int (*do_ioctl)(struct hnae3_handle *handle,
+			struct ifreq *ifr, int cmd);
+#endif
 };
 
 struct hnae3_dcb_ops {
